src/SFML/System/FileInputStream.cpp

```diff
@@ -78,39 +78,39 @@ bool FileInputStream::open(const std::filesystem::path& filename)
 
 
 ////////////////////////////////////////////////////////////
-std::int64_t FileInputStream::read(void* data, std::int64_t size)
+std::optional<std::size_t> FileInputStream::read(void* data, std::size_t size)
 {
     if (!m_file)
-        return -1;
+        return std::nullopt;
 #ifdef SFML_SYSTEM_ANDROID
     return m_file->read(data, size);
 #else
-    return static_cast<std::int64_t>(std::fread(data, 1, static_cast<std::size_t>(size), m_file.get()));
+    return std::fread(data, 1, size, m_file.get());
 #endif
 }
 
 
 ////////////////////////////////////////////////////////////
-std::int64_t FileInputStream::seek(std::int64_t position)
+std::optional<std::size_t> FileInputStream::seek(std::size_t position)
 {
     if (!m_file)
-        return -1;
+        return std::nullopt;
 #ifdef SFML_SYSTEM_ANDROID
     return m_file->seek(position);
 #else
     if (std::fseek(m_file.get(), static_cast<long>(position), SEEK_SET))
-        return -1;
+        return std::nullopt;
 
     return tell();
 #endif
 }
 
 
 ////////////////////////////////////////////////////////////
-std::int64_t FileInputStream::tell()
+std::optional<std::size_t> FileInputStream::tell()
 {
     if (!m_file)
-        return -1;
+        return std::nullopt;
 #ifdef SFML_SYSTEM_ANDROID
     return m_file->tell();
```

Comment:
This is a bug since it may return `-1` which when implicitly converted to a `std::optional<std::size_t>` leads to the wrong result. The optional needs to be null when `-1` is returned.